update docker-py to 7.1.0

thaJeztah opened the pull request:
- fixes https://github.com/moby/moby/issues/46967
- [x] depends on / stacked on https://github.com/moby/moby/pull/47091
- relates to https://github.com/moby/moby/pull/46965
- relates to https://github.com/moby/moby/pull/46966
- relates to https://github.com/docker/docker-py/pull/3203

### update docker-py to 7.1.0

full diff: https://github.com/docker/docker-py/compare/7.0.0...7.1.0

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
update docker-py to 7.1.0

full diff: https://github.com/docker/docker-py/compare/7.0.0...7.1.0

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

vvoland commented:
We should undraft this if we want to deprecate Container related fields in image inspect in v26: https://github.com/docker/cli/pull/4893

vvoland commented:
Hmm looks like we have the skips but they don't work? 🤔 

At least https://github.com/moby/moby/pull/47430 shows related docker-py failures.

thaJeztah commented:
> they're only for the containerd integration.

Ah, right, I guess we could still skip unconditionally. That said; not 100% against updating to "master"; mostly was trying to stay on tagged versions, but it's not _critical_

krissetto commented on hack/make/test-docker-py:
```diff
@@ -55,8 +55,9 @@ PY_TEST_OPTIONS="$PY_TEST_OPTIONS --deselect=tests/integration/api_image_test.py
+			# TODO(thaJeztah): set a default VERSION in the tests/integration Dockerfile in docker-py or set SETUPTOOLS_SCM_PRETEND_VERSION conditionally
 			# shellcheck disable=SC2086
-			exec docker build ${build_opts} -t "${docker_py_image}" -f tests/Dockerfile "https://github.com/docker/docker-py.git#${DOCKER_PY_COMMIT}"
+			exec docker build ${build_opts} -t "${docker_py_image}" --build-arg VERSION="${DOCKER_PY_COMMIT}" -f tests/Dockerfile "https://github.com/docker/docker-py.git#${DOCKER_PY_COMMIT}"
```
Oh no, yet another env var whose value then gets used for `SETUPTOOLS_SCM_PRETEND_VERSION` ? 🙈

thaJeztah replied:
Yup; looks like it was there before, but at the time it had a default value in the Dockerfile (99.0.0?).

That default was removed, so now it must be passed. We should fix that, because I think we should try to make any Dockerfile have sensible defaults, so that they work without passing build-args (only in exceptional cases where a build-arg MUST be set).

thaJeztah commented:
Rebased, and updated the TODO's to include a link to https://github.com/docker/docker-py/pull/3267, which should allow removing the `VERSION` build-arg.
